Lib/unittest/mock.py

```diff
@@ -3003,6 +3005,59 @@ def __set__(self, obj, val):
         self(val)
 
 
+class ThreadingMock(MagicMock):
+    """
+    A mock that can be used to wait until on calls happening
+    in a different thread.
+    """
+
+    def __init__(self, *args, **kwargs):
+        _safe_super(ThreadingMock, self).__init__(*args, **kwargs)
+        self.__dict__["_event"] = threading.Event()
+        self.__dict__["_expected_calls"] = []
+        self.__dict__["_events_lock"] = threading.Lock()
+
+    def __get_event(self, expected_args, expected_kwargs):
+        with self._events_lock:
+            for args, kwargs, event in self._expected_calls:
+                if (args, kwargs) == (expected_args, expected_kwargs):
+                    return event
```

Comment:
You can just wait on them, that way you know which one failed easily:
```python
block_until_any_call("arg1", kwarg2="no")
block_until_any_call("arg2", kwarg2="yes")
```

If you want to specify order, you can use assert:

```python
block_until_any_call("arg2", kwarg2="yes")
assert_calls(call("arg1", kwarg2="no"), call("arg2", kwarg2="yes"))
```

My worry of implementing `wait_until_calls` is that it will continue waiting until the calls happen in the specific order.

If we want full completeness, I'd suggest also implementing `block_until_call`, to wait for *the first call* to match the specified arguments.